desafio-11/WhoisBsa/javascript/index.ts

```diff
@@ -0,0 +1,70 @@
+// Primes in pi - WhoisBsa
+import fs from 'fs';
+
+const data = fs.readFileSync('./data.txt', 'utf8');
+const numbers = data.slice(2);
+const primeList = new Set<number>();
+
+let primeSequence = '';
+let maxSequence = '';
+let currentSequence = '';
+let currentNumber = '';
+let numberIndex = 0;
+
+const minPrime = 2;
+const maxPrime = 9973;
+const isPrimeNumber = (number: number) => {
+  if (number < 2 || (number % 2 == 0 && number > 2)) return false;
+
+  const s = Math.sqrt(number);
+  for (let i = 3; i <= s; i += 2)
+    if (number % i === 0) return false;
+
+  primeList.add(number);
+};
+
+isPrimeNumber(minPrime);
```

Comment:
Aqui não sei se era algum teste que ficou pra traz. Mas vc roda só para o minPrime a funçao.
isPrimeNumber(minPrime);

Reply:
Sim, porque como no `for` eu percorro apenas os ímpares maiores que 3, eu tenho que adicionar na lista o 2 primeiro.

Reply:
A abordagem atual pode ser aprimorada. Ao criar uma função, estamos encapsulando um trecho de código que pode ser reutilizado por você e por outras pessoas em diferentes contextos.

É importante que uma função faça apenas uma coisa. No caso atual, a função retorna falso quando o número não é primo e adiciona o número a um conjunto quando é primo.

Seria mais apropriado que a função se concentrasse apenas na verificação caso primo e retornasse true ou false com base nessa verificação.

Minha sugestão é que você modifique a função tratando o caso do número 2 internamente e realize nela apenas a verificação. Para qualquer primo true, não primo false.

Movendo a lógica de adição para fora da função. Por exemplo:

if (isPrimeNumber(i)) {
    primeList.add(i);
}